engine/engine/content/builtins/scripts/edn.lua

```diff
@@ -68,38 +55,34 @@ local function encode_number(val)
 end
 
 local function encode_string(val)
-  return '"' .. val:gsub('[%z\1-\31\\"]', escape_char) .. '"'
+  return '"'..val:gsub('[%z\1-\31\\"]', escape_char)..'"'
 end
 
-local function encode_as_tagged_string(tag, val)
-  return tag .. encode_string(str(val))
+local function encode_table(val)
+  return string.format('#lua/ref"%p"', val)
 end
 
-local function make_tagged_string_encoder(tag)
-  return function(val)
-    return encode_as_tagged_string(tag, val)
-  end
+function encode_userdata(val)
+  return '#lua/userdata"'..str(val)..'"'
 end
 
-local function get_table_ref(t)
-  return get_prefixed_addr("table: ", t)
+function encode_function(val)
+  return '#lua/function"'..str(val)..'"'
```

Comment:
We know how function serialised tostring so, it's not necessary to call heavy `encode_string` function here

The same for `encode_thread` but I'm not sure about `encode_userdata`, what do you think?